src/main/java/sirius/kernel/xml/Outcall.java

```diff
@@ -337,6 +391,41 @@ public String getHeaderField(String name) {
         return connection.getHeaderField(name);
     }
 
+    /**
+     * Returns the response header with the given name as an optional {@link LocalDateTime}.
+     *
+     * @param name the name of the header to fetch
+     * @return the date of the given header wrapped in an Optional or empty if the field does not exists or can not be parsed as date
+     */
+    public Optional<LocalDateTime> getHeaderFieldDate(String name) {
+        long timestamp = connection.getHeaderFieldDate(name, -1);
+        if (timestamp >= 0) {
+            return Optional.of(Instant.ofEpochMilli(timestamp).atZone(ZoneId.systemDefault()).toLocalDateTime());
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Tries to parse a file name from the content disposition header.
+     * <p>
+     * The format of the header is defined here: http://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html
+     * This header provides a filename for content that is going to be downloaded to the file system.
+     *
+     * @return an Optional containing the file name given by the header, or Optional.empty if no file name is given
+     */
+    public Optional<String> parseFileNameFromContentDisposition() {
+        try {
+            Matcher matcher =
+                    CONTENT_DISPOSITION_FILENAME_PATTERN.matcher(connection.getHeaderField(HEADER_CONTENT_DISPOSITION));
+            if (matcher.find()) {
+                return Optional.ofNullable(matcher.group(2));
+            }
+        } catch (IllegalStateException | IndexOutOfBoundsException e) {
```

Comment:
why would these occur? the pattern is guaranteed to have 3 groups and the state is consistent if find returns true (?)